### PR TITLE
Setup project with redux-react and redux toolkit

### PR DESCRIPTION
**Pull request by Esmatullah12**
for this part of the project I worked on the following things
- Understand the concept of state management.
- Use store, actions, and reducers in React.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const bookSlice = createSlice({
+  name: 'books',
+  initialState: [],
+  reducers: {
+    addBook: () => {
+
+    },
+    removeBook: () => {
+
+    },
```
Comment (Owusu-Desmond) [inline src/redux/books/booksSlice.js]: - [ ] Please make sure to complete the `addBook` and `removeBook` reducers. These functions will require the state and action as parameters. The `addBook` function is responsible for adding a book to the `books` array in the state, while the `removeBook` function will remove a book from the `books` array based on its `id`.

File: src/redux/store.js
Comment (Owusu-Desmond) [inline src/redux/store.js:1]: - [ ] For your store configuration, please follow these steps:
  
  - To begin, import `configureStore` from `@reduxjs/toolkit`.
  - Next, import `books` from `src/redux/books`.
  - Then, import `categories` from `src/redux/categories`.
  - Use the `configureStore` method to create a reducer.
  - Finally, add `books` and `categories` to the `configureStore` method and export the store.

  This implementation is similar to what you need and should work effectively.
  
    ```jsx
  // ./src/redux/store.js
  
  import { configureStore } from "@reduxjs/toolkit"
  import booksReducer from "./books/booksSlice"
  import categoriesReducer from "./categories/categoriesSlice"
  
  const store = configureStore({
    reducer: { books: booksReducer, categories: categoriesReducer },
  })
  
  export default store
  ```

File: src/redux/categories/categoriesSlice.js
Comment (Owusu-Desmond) [inline src/redux/categories/categoriesSlice.js:1]: - [ ] To ensure proper functionality, please follow these steps when creating your categoriesSlice

    - Import `createSlice` from `@reduxjs/toolkit`
    - Create an `initialState` with a `categories` state value that is set to an empty array
    - Create a single reducer called `checkStatus`
    - Export `categoriesSlice.reducer` by default
    - Export `checkStatus` as an action using the `export` keyword

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @Esmatullah12 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Esmatullah12,

✔️ Implementation of store using redux toolkit

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**